src/Bicep.Core/Intermediate/CrossModuleSentinelVisitor.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Deployments.Expression.Intermediate;
+
+namespace Bicep.Core.Intermediate
+{
+    public class CrossModuleSentinelVisitor : TemplateLanguageExpressionVisitor
```

Comment:
[nit]: I would make this a `private` class internal to `WhatIfShortCircuitingRule`. Not sure there's much potential for reuse here.